libs-haskell/da-hs-base/src/DA/Service/Logger/Impl/GCP.hs

```diff
@@ -399,11 +400,11 @@ today = Time.utctDay <$> getCurrentTime
 maxDataPerDay :: SentBytes
 maxDataPerDay = SentBytes (8 * 2 ^ (20 :: Int))
 
--- | The DAML home folder, getting this ensures the folder exists
-getDamlDir :: Maybe FilePath -> IO FilePath
-getDamlDir damlPathM = do
-    dh <- lookupEnv damlPathEnvVar
-    dir <- case damlPathM <|> dh of
+-- | The DAML cache folder, getting this ensures the folder exists and is writable.
```

Comment:
How is writability ensured?

Reply:
It checks the permissions now and returns `Nothing` if we end up with a read-only cache.